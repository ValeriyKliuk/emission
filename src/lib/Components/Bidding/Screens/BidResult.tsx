import React from "react"
import styled from "styled-components/native"

import { Fonts } from "../../../data/fonts"
import Serif from "../../Text/Serif"
import { Button } from "../Components/Button"
import { Container } from "../Components/Container"
import { Timer } from "../Components/Timer"
import { Title } from "../Components/Title"

interface BidResultProps {
  winning: boolean
}

export class BidResult extends React.Component<BidResultProps> {
  render() {
    if (this.props.winning) {
      return (
        <CenteringContainer>
          <Icon source={require("../../../../../images/checkbox.png")} />
          <Title>You're the highest bidder</Title>

          <TimeLeft>Time left</TimeLeft>
          <Timer timeLeftInMilliseconds={1000 * 60 * 20} />
        </CenteringContainer>
      )
    } else {
      return (
        <Container>
          <CenteringContainer>
            {/* TODO: Use the x icon instead of √ */}
            <Icon source={require("../../../../../images/checkbox.png")} />
            <Title>Your bid wasn’t high enough</Title>

            <StyledText>Another bidder placed a higher max bid or the same max bid before you did.</StyledText>
            <StyledText>Increase your max bid to take the lead.</StyledText>

            <TimeLeft>Time left</TimeLeft>
            <Timer timeLeftInMilliseconds={1000 * 60 * 20} />

            <Divider />

            <Serif>Current bid</Serif>
            <Title>$45,000</Title>
          </CenteringContainer>

          <Button text="Bid again" onPress={() => null} />
        </Container>
      )
    }
  }
}

const CenteringContainer = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`

const Icon = styled.Image`
  width: 20px;
  height: 20px;
  margin: 10px;
`

const StyledText = styled.Text`
  margin: 10px;
  font-family: ${Fonts.GaramondRegular};
  font-size: 16px;
  text-align: center;
  color: #666666;
`

const TimeLeft = styled.Text`
  font-size: 12px;
  color: #666666;
  margin-top: 10px;
`

const Divider = styled.View`
  border-bottom-color: #e5e5e5;
  border-bottom-width: 1px;
  margin: 30px 0 20px;
  width: 100%;
`
