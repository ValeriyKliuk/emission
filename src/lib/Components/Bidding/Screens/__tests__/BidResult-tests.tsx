import React from "react"
import "react-native"

import * as renderer from "react-test-renderer"
import { BidResult } from "../BidResult"

it("renders winning screen properly", () => {
  const bg = renderer.create(<BidResult winning />).toJSON()
  expect(bg).toMatchSnapshot()
})

it("renders not highest bid screen properly", () => {
  const bg = renderer.create(<BidResult winning={false} />).toJSON()
  expect(bg).toMatchSnapshot()
})
