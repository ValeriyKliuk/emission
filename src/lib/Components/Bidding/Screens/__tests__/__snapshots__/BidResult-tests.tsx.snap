// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders not highest bid screen properly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
        "marginBottom": 20,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../images/checkbox.png",
        }
      }
      style={
        Array [
          Object {
            "height": 20,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": 20,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "AGaramondPro-Regular",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Your bid wasn’t high enough
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#666666",
            "fontFamily": "AGaramondPro-Regular",
            "fontSize": 16,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Another bidder placed a higher max bid or the same max bid before you did.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#666666",
            "fontFamily": "AGaramondPro-Regular",
            "fontSize": 16,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Increase your max bid to take the lead.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#666666",
            "fontSize": 12,
            "marginTop": 10,
          },
          undefined,
        ]
      }
    >
      Time left
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "500",
          },
          undefined,
        ]
      }
    >
      0
      d 
      0
      h 
      20
      m 
      0
      s
    </Text>
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#e5e5e5",
            "borderBottomWidth": 1,
            "marginBottom": 20,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 30,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          undefined,
          Object {
            "fontFamily": "AGaramondPro-Regular",
          },
        ]
      }
    >
      Current bid
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "AGaramondPro-Regular",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      $45,000
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 1)",
        "borderColor": "rgba(0, 0, 0, 1)",
        "borderWidth": 1,
        "height": 46,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 12,
          },
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "AGaramondPro-Regular",
            "fontSize": 14,
            "opacity": 1,
          },
          Object {
            "fontFamily": "AvantGardeGothicITC",
          },
        ]
      }
    >
      BID AGAIN
    </Text>
  </View>
</View>
`;

exports[`renders winning screen properly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../images/checkbox.png",
      }
    }
    style={
      Array [
        Object {
          "height": 20,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "width": 20,
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "AGaramondPro-Regular",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    You're the highest bidder
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#666666",
          "fontSize": 12,
          "marginTop": 10,
        },
        undefined,
      ]
    }
  >
    Time left
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 14,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    0
    d 
    0
    h 
    20
    m 
    0
    s
  </Text>
</View>
`;
